pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,670 @@
+                                Zl: complex | FrequencyData) -> FrequencyData:
+        """Returns the transfer function Q1_out / Q2_in."""
+        denominator = (self.C * Zl + self.D)
+        nominator = Zl*FrequencyData(np.ones_like(denominator.freq),
+                                     self.frequencies)
+
+        # Admittance form for Zl = inf
+        idx_inf, __, __  = self._check_for_inf(Zl)
+        nominator.freq[idx_inf] = 1
+        denominator.freq[idx_inf] = (self.C + self.D / Zl).freq[idx_inf]
+
+        # In cases where the denominator is zero, e.g. Zl = inf & C = 0,
+        # are related to non-physical cases (e.g. undefined current/voltage)
+        # => NaN
+        denominator.freq[denominator.freq == 0] = np.nan
+        return nominator / denominator
+
+    @classmethod
+    def create_identity(tmat, frequencies, abcd_cshape = ()):
+        r"""Creates an object with identity matrix entries (bypass).
+
+        See Equation (2-7) in Table I of Reference [1]_:
+
+        .. math::
+            T = \begin{bmatrix}
+                1 & 0 \\
+                0 & 1
+                \end{bmatrix}
+
+        Parameters
+        ----------
+        frequencies : array_like
+            The frequency sampling points in Hz.
+        abcd_cshape : tuple | int, optional
+            Shape of additional channel dimensions besides (2,2).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        tmat : TransmissionMatrix
+            A TransmissionMatrix object that contains 2x2 identity matrices.
+
+        """
+        A = np.ones((*abcd_cshape, np.array(frequencies).size))
+        (B,C,D) = (0,0,1)
+        return tmat.from_abcd(A, B, C, D, frequencies)
+
+    @classmethod
+    def create_series_impedance(tmat, impedance: FrequencyData,
+                                abcd_cshape=()):
+        r"""Creates a transmission matrix representing a series impedance.
+
+        This means the impedance is connected in series with a potential load
+        impedance. See Equation (2-8) in Table I of Reference [1]_:
+
+        .. math::
+            T = \begin{bmatrix}
+                1 & Z \\
+                0 & 1
+                \end{bmatrix}
+
+        Parameters
+        ----------
+        impedance : FrequencyData
+            The impedance data of the series impedance.
+        abcd_cshape : tuple | int, optional
+            Shape of additional channel dimensions besides (2,2).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        tmat : TransmissionMatrix
+            A TransmissionMatrix representing a series impedance.
+
+        """
+        if impedance.cshape != (1,):
+            raise ValueError("Number of channels for 'impedance' must be 1.")
+
+        (Z, frequencies) = (impedance.freq[0], impedance.frequencies)
+        Z = np.zeros((*abcd_cshape, np.array(frequencies).size)) + Z
```

Comment:
Is it possible to work with https://numpy.org/doc/stable/reference/generated/numpy.broadcast_to.html instead?